File: gym_management/gym_module/report/fitness_journey_report/fitness_journey_report.js
frappe.query_reports["Fitness Journey Report"] = {
	"filters": [
	    {
		  fieldname: "member",
		  label: __("Member"),
		  fieldtype: "Link",
		  options: "Gym Member",
		  reqd: 1,
		  default: "",
		  get_query: function() {
			return {
			    filters: {
				  "status": "Active"
			    }
			};
		  }
	    }
	],
  
	// Optional: Customize chart colors and behavior after loading
	onload: function(report) {
	    frappe.msgprint({
		  message: __("📊 Select a Member to view their Fitness Journey."),
		  title: __("Welcome!"),
		  indicator: "blue"
	    });
	},
  
	formatter: function(value, row, column, data, default_formatter) {
	    // Highlight BMI values
	    if (column.fieldname === "bmi" && value) {
		  let bmi = parseFloat(value);
		  let color = "green";
		  if (bmi < 18.5) color = "orange"; // underweight
		  else if (bmi > 25) color = "red"; // overweight
		  value = `<span style="color:${color}; font-weight:bold;">${value}</span>`;
	    }
	    return default_formatter(value, row, column, data);
	}
  };
  

File: gym_management/gym_module/report/fitness_journey_report/fitness_journey_report.py


import frappe

def execute(filters=None):
    columns = [
        {"label": "Date", "fieldname": "date", "fieldtype": "Date", "width": 120},
        {"label": "Weight (kg)", "fieldname": "weight", "fieldtype": "Float", "width": 100},
        {"label": "Calories Intake", "fieldname": "calories_intake", "fieldtype": "Float", "width": 120},
        {"label": "BMI", "fieldname": "bmi", "fieldtype": "Float", "width": 120}
    ]

    data = frappe.get_all("Gym Progress Tracker",
        filters={"member": filters.get("member")},
        fields=["date", "weight", "calories_intake", "bmi"],
        order_by="date asc"
    )

    chart = {
        "data": {
            "labels": [d["date"] for d in data],
            "datasets": [
                {"name": "Weight (kg)", "values": [d["weight"] for d in data]},
                {"name": "Calories", "values": [d["calories_intake"] for d in data]},
                {"name": "BMI", "values": [d["bmi"] for d in data]}
            ]
        },
        "type": "line",
        "height": 300
    }

    return columns, data, None, chart
